std/range/package.d

```diff
@@ -2427,7 +2427,16 @@ if (Rs.length > 1 && allSatisfy!(isInputRange, staticMap!(Unqual, Rs)))
         }
     }
 
-    return Result(rs, 0);
+    size_t firstNonEmpty = () {
+        static foreach (i; 0 .. Rs.length)
+        {
+            if (!rs[i].empty)
+                return i;
+        }
+        return size_t.max;
+    }();
```

Comment:
Why is this using a function literal? It made me think this was a static variable, but it's actually a local variable.

Reply:
There's no `static break`, so I'm using `return` to exit the loop early. If you think it's too confusing we can change it to this:

```d
    size_t firstNonEmpty = size_t.max;
    static foreach (i; 0 .. Rs.length)
    {
        if (firstNonEmpty == size_t.max && !rs[i].empty)
            firstNonEmpty = i;
    }
```

Reply:
It's mostly confusing on GitHub's viewer where I can't easily see what scope I'm in. I guess it's fine when reading in an editor.

Reply:
That's a good enough reason for me. I'll go ahead and make the change.